### ISSUE TITLE
How to successfully clear doctrine cache?

### ISSUE THREAD

**Issue by fico7489** [doctrine/doctrine2#11133]
I have 1000 items in database with ids 1..1000 and I am doing foreach with batches

```
$batches = [
   [
        1,2,3,...100
   ],
   [
       101, 102, 103,...,200
   ],
   ...
   [
      900, 901,902,...1000 
   ]
]


 foreach ($batches as $batchIds) {
     $items = $this->entityManager->getRepository($className)->findBy(['id' => $batchIds]);
     
     foreach ($items as $item){
            $repository = $this->entityManager->getRepository($className);
            $item = $repository->findOneBy(['id' => $item->getId()]);

            //$this->entityManager->refresh($item);
            
            dump(round((memory_get_usage() / 1024) / 1024, 2) . 'M');
        }
 }
```
     
I am dumping memory usage after each batch, memory_usage for last batch is **60MB**, but when I uncomment :

`$this->entityManager->refresh($item);`

it is **85MB** for last batch.

Does anyone know why refresh() is using 30% more memory for the same entity?

It is just example code, I am fetching and refreshing each entity because that code is sometimes called from queue and sometimes in sync, but that is not important, I am just wondering why refresh() is using so much memory and how I can clear memory after each batch. 

I tried with adding this after each batch:

```
$cache = $this->entityManager->getConfiguration()->getQueryCache();
$cache->clear();

$cache = $this->entityManager->getConfiguration()->getMetadataCache();
$cache->clear();

$cache = $this->entityManager->getConfiguration()->getResultCache();
$cache->clear();

gc_collect_cycles();

$this->entityManager->flush();
$this->entityManager->clear();

$this->entityManager->getUnitOfWork()->clear();
```

result is now **35MB** without refresh(), and **65MB** with refresh() for last batch.

To sum up:

without refresh, without clearing cache: **60MB**
with refresh, without clearing cache: **85MB**
without refresh, with clearing cache: **35MB**
with refresh, with clearing cache: **65MB**

That issue causes me problems for larger set of data.

Does anyone know what kind of cache refresh() is generating and how I can clear that cache between batches?

**Comment by fico7489** [comment]
the fix was to run symfony app with:

APP_DEBUG=0

**Comment by fico7489** [comment]
the fix was to run symfony app with:

APP_DEBUG=0
